### PR TITLE
[WA-28]: Added current weather and current location API

### PR DESCRIPTION
**Pull request by Anurov-Prihodko**

### CODE REVIEW

File: src/app/details/details.component.ts [hunk 1]
```diff
@@ -9,9 +11,45 @@ import { CityWeatherInfo } from '../shared/interfaces/city-weather-info.interfac
...
+    temp: '',
+    isFavorite: false,
+  };
+
+  position: any = {};
+
+  constructor(private weatherServise: WeatherService) {}
+
+  async ngOnInit() {
+    if (navigator.geolocation) {
```
Comment (dmitry-chirva) [inline src/app/details/details.component.ts:25]: this.geolocationService.getPosition()
.pipe(
  swithMap(({ area }) => {
    return this.weatherServise.getCurrentWeatherByArea(area )
  })
)
.subscribe((weather => {
  this.weatherInfo = weather;
}))

File: src/app/footer/footer.component.ts [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
 })
-export class FooterComponent {
+export class FooterComponent implements OnInit {
+  constructor(private weatherServise: WeatherService) {}
+
+  getYear: string = '';
+
+  ngOnInit() {
+    this.weatherServise.getWeather().subscribe((res) => {
+      const year = res.location.localtime.split('').slice(0, 4).join('');
```
Comment (dmitry-chirva) [inline src/app/footer/footer.component.ts:16]: new Date().getYear()

File: src/app/details/details.component.ts (hunk 1 above)
Comment (dmitry-chirva) [reply src/app/details/details.component.ts:25]: getCurrentWeather() {
reuturn this.weatherAPI.getweathe().pipe(
   swithMap
   ...
   map()
)
}


homeService.getCurrentWeather()
.subscribe()

File: src/app/details/details.component.ts [hunk 3]
```diff
@@ -9,9 +11,45 @@ import { CityWeatherInfo } from '../shared/interfaces/city-weather-info.interfac
...
+
+          this.weatherInfo.date = `${month} ${date}th, ${year}`;
+          this.weatherInfo.temp = `${Math.floor(response.current.temp_c)} °С`;
+          this.weatherInfo.city = `${response.location.name}, ${response.location.country}`;
+        });
+    } else alert('Geolocation is not supported!');
+  }
+
+  getCoords(options?: PositionOptions): Promise<any> {
+    return new Promise((resolve, reject) =>
```
[the remark this answers is not in the document]
Comment (Anurov-Prihodko) [reply src/app/details/details.component.ts:51]: use is a method?

[the remark this answers is not in the document]
Comment (Anurov-Prihodko) [reply src/app/details/details.component.ts:51]: Done

File: src/core/api/weather/weather.service.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    days = 1,
+    aqi = 'no',
+    alerts = 'no'
+  ): Observable<ForecastI> {
+    return this.http.get<ForecastI>(
+      `${this.BASE_URL}forecast.json?key=${this.API_KEY}&q=${town}&days=${days}&aqi=${aqi}&alerts=${alerts}`
+    );
+  }
+
+  getCurrentWeather(position: any): Observable<CurrentLocationI> {
```
Comment (dmitry-chirva) [inline src/core/api/weather/weather.service.ts:28]: Change any to interface

File: src/core/api/weather/weather.service.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  providedIn: 'root',
+})
+export class WeatherService {
+  readonly BASE_URL = 'http://api.weatherapi.com/v1/';
+  readonly API_KEY = 'b88614b3fb684e8b996104153220302';
+
+  constructor(private http: HttpClient) {}
+
+  getForecastWeather(
+    town = 'Kiev',
```
Comment (dmitry-chirva) [inline src/core/api/weather/weather.service.ts:18]: change town to city

File: src/core/api/weather/types-weather.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+type CurrentWeatherData = {
```
Comment (dmitry-chirva) [inline src/core/api/weather/types-weather.ts:1]: export type

Comment (dmitry-chirva) [reply src/core/api/weather/types-weather.ts:1]: weather.type.ts

File: src/core/api/weather/geo-location.service.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        },
+        (error) => {
+          obs.error(error);
+        }
+      );
+    });
+  }
+}
+
+// getCoords(): Promise<any> {
```
Comment (dmitry-chirva) [inline src/core/api/weather/geo-location.service.ts:25]: remove comments

File: src/core/api/home/home.service.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  providedIn: 'root',
+})
+export class HomeService {
+  constructor(
+    private weatherService: WeatherService,
+    private geolocationService: GeoLocationService
+  ) {}
+
+  getCurrentWeatherHome(): any {
+    if (navigator.geolocation) {
```
Comment (dmitry-chirva) [inline src/core/api/home/home.service.ts:17]: remove condition

File: src/app/shared/interfaces/search-info.interfaces.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+export interface Location {
+  name: string;
+  region: string;
+  country: string;
+  localtime: string;
+}
+
+export interface Current {
```
Comment (dmitry-chirva) [inline src/app/shared/interfaces/search-info.interfaces.ts:13]: CurrentTemp

File: src/app/home/home.component.ts [hunk 10]
```diff
@@ -1,16 +1,39 @@
...
+    temp: '',
+    isFavorite: false,
+  };
+
+  constructor(private homeService: HomeService) {}
+
+  ngOnInit() {
+    this.homeService
+      .getCurrentWeatherHome()
+      .subscribe((response: CurrentWeatherData) => {
```
Comment (dmitry-chirva) [inline src/app/home/home.component.ts:25]: create transformation service for change be data to fe data

File: src/core/api/weather/weather.service.ts (hunk 4 above)
Comment (Anurov-Prihodko) [reply src/core/api/weather/weather.service.ts:28]: Done

File: src/core/api/weather/weather.service.ts (hunk 5 above)
Comment (Anurov-Prihodko) [reply src/core/api/weather/weather.service.ts:18]: Done

File: src/core/api/weather/types-weather.ts (hunk 6 above)
Comment (Anurov-Prihodko) [reply src/core/api/weather/types-weather.ts:1]: Done

File: src/core/api/weather/geo-location.service.ts (hunk 7 above)
Comment (Anurov-Prihodko) [reply src/core/api/weather/geo-location.service.ts:25]: done

File: src/core/api/home/home.service.ts (hunk 8 above)
Comment (Anurov-Prihodko) [reply src/core/api/home/home.service.ts:17]: Done

File: src/app/shared/interfaces/search-info.interfaces.ts (hunk 9 above)
Comment (Anurov-Prihodko) [reply src/app/shared/interfaces/search-info.interfaces.ts:13]: Done

**Pull request merged**